paymentsheet/src/main/java/com/stripe/android/lpmfoundations/paymentmethod/PaymentMethodMetadata.kt

```diff
@@ -69,7 +69,7 @@ internal data class PaymentMethodMetadata(
             stripeIntent.isLiveMode &&
                 stripeIntent.unactivatedPaymentMethods.contains(it.type.code)
         }.filter { paymentMethodDefinition ->
-            sharedDataSpecs.firstOrNull { it.type == paymentMethodDefinition.type.code } != null
+            paymentMethodDefinition.uiDefinitionFactory().canBeDisplayedInUi(paymentMethodDefinition, sharedDataSpecs)
```

Comment:
This should be the only behavior change in the whole PR, with an associated test at the bottom of the diff.